backend/core/src/main/java/com/feed_grabber/core/report/ReportService.java

```diff
@@ -54,4 +61,32 @@ public String generateReport(UUID requestId) throws NotFoundException, JsonProce
     public String getReport(UUID requestId) {
         return template.getForObject(EP +"/report/"+requestId, String.class);
     }
+
+    public void hasAccess(UUID requestId, UUID userId, String role) {
+        if (role.equals(ROLE_COMPANY_OWNER) || role.equals(ROLE_HR)) {
+            return;
+        }
+        Optional<Request> request = requestRepository.findByIdAndTargetUser(requestId, userId);
+        if (request.isEmpty() || !request.get().getSendToTarget()) {
+            throw new AccessDeniedException("You have not enough permissions to view this report");
+        }
+    }
```

Comment:
This method checks if the user has access to the report.
We grant access only in three cases:
1-2) if the user is a company owner or HR
3) if the user is a target user of request and field `sendToRequest` is true